app/controllers/events_controller.rb

```diff
@@ -49,6 +49,7 @@ def create
 
   # PATCH/PUT /events/1
   def update
+    @event.invalidate_schedule if event_params[:has_place_3_match].to_i.zero? == @event.has_place_3_match
```

Comment:
You sure this should happen here? Not in an `after_commit` hook on the model? You cannot be sure if the model will actually be saved, and you delete all matches. Even those with results.

Reply:
Sounds good. How will this work?

Reply:
Checkout: https://apidock.com/rails/ActiveRecord/Transactions/ClassMethods/after_commit
Rails offers mutltiple hook methods so that you can perform actions on the model after specific actions (in this case update, create or destroy).
Goal should be to keep the controller somwhat lean ;)

Reply:
Seems, like we don't have the information we need in a after_commit hook.